libbeat/autodiscover/providers/kubernetes/node.go

```diff
@@ -109,7 +109,10 @@ func NewNodeEventer(
 // OnAdd ensures processing of node objects that are newly created
 func (n *node) OnAdd(obj interface{}) {
 	n.logger.Debugf("Watcher Node add: %+v", obj)
-	n.emit(obj.(*kubernetes.Node), "start")
+	err:= n.emit(obj.(*kubernetes.Node), "start")
```

Comment:
The issue is coming from here: emit is not returning anything and therefore you can't catch an error from there.